v21.1/developer-guide-overview.md

```diff
@@ -4,32 +4,65 @@ summary: An overview of common tasks that come up when you build an application
 
-## Sample Apps and Tutorials
+## CockroachDB basics
 
-For simple but complete sample apps and tutorials, see [Hello World](hello-world-example-apps.html).
+Before you start using CockroachDB, you should understand a couple important mechanics of the database:
 
-For more advanced sample apps and tutorials, see [Build a Spring App with CockroachDB](build-a-spring-app-with-cockroachdb-jdbc.html) and [Develop and Deploy a Multi-Region Web Application](multi-region-overview.html).
+- [How transactions work in CockroachDB](#how-transactions-work-in-cockroachdb)
+- [How applications interact with CockroachDB](#how-applications-interact-with-cockroachdb)
 
-{% include cockroach_u_pydev.md %}
+Note that the sections that follow were written for the purposes of orienting application developers. For more detailed technical documentation on CockroachDB's architecture, see our [architecture documentation](architecture/overview.html).
+
+### How transactions work in CockroachDB
+
+CockroachDB is designed to make your data scalable and consistent. All stored records are replicated across a distributed deployment of database instances, and all database transactions committed in CockroachDB satisfy [ACID properties](https://en.wikipedia.org/wiki/ACID) and the [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem). This means that when you store data in CockroachDB, the data will be valid and anomaly-free, even in the event of a system error or power failure.
+
+To guarantee that database operations are [atomic](https://en.wikipedia.org/wiki/Atomicity_(database_systems)) (the "A" of the [ACID properties](https://en.wikipedia.org/wiki/ACID)), CockroachDB executes all statements in the context of an atomic [database transaction](https://en.wikipedia.org/wiki/Database_transaction). If a transaction succeeds, all data mutations are applied together with virtual simultaneity. If any part of a transaction fails, the entire transaction is aborted, and the database is left unchanged.
+
+Managing transactions is an important part of CockroachDB application development. When creating your application's persistence layer, you must understand how CockroachDB interacts with your application framework's transaction management system. You should also have some knowledge of CockroachDB's [transaction syntax](transactions.html#syntax). We go into more detail about transaction management in [Manage Transactions](transactions.html).
+
+#### Serializability and transaction contention
+
+CockroachDB guarantees [`SERIALIZABLE`](https://en.wikipedia.org/wiki/Serializability) transaction [isolation](https://en.wikipedia.org/wiki/Isolation_(database_systems) (the "I" of ACID semantics). If transactions are executed concurrently, the final state of the database will appear as if the transactions were executed serially. `SERIALIZABLE` isolation, the strictest level of isolation, provides the highest level data consistency and protects against concurrency-based attacks.
```

Comment:
should it say "highest level of data consistency" ?

maybe mention that it protects from bugs as well as attacks ?